tests/cpp17/stringify_util.h

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2020 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This contains some utilities/examples for how to leverage the static reflec-
```

Comment:
This is great for testing.. I wonder if some of it can be generic enough that its worth sticking in `include/flatbuffers` ? Wonder what happens if someone actually wants to use this functionality, right now they'd probably to copy this.

Reply:
Yes this is a standalone utility that someone could include and use in their own project if they are compiling with C++17.  If you like I can move it into the include/flatbuffers folder so that they can include it more easily.

Reply:
@vglavnyy do you think that makes sense?

Reply:
I don't mind. However, we already have a lot of changes in C++ code: `optional`, `span`, `--cpp-std c++11`.
Maybe move it into the include folder after release?

Reply:
Ok for now I will just leave it in the test folder, and we can move it at a later time.

Another thing to think about -- it is possible to parse this string and produce a real object with roughly the same amount of code/methodology.  Perhaps we might want to include that as well when the time comes.

Another use case of this reflection is that we can provide a function called `FlatbuffersEquals` that will compare two Flatbuffers of the same type and tell you if they are equal or not.  It will do this by iterating through the fields and comparing them, again using the static reflection.  I believe traditionally this has not been possible to compare two buffers right?